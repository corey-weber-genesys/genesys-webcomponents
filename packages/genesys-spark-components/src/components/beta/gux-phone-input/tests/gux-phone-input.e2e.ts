import { a11yCheck, newSparkE2EPage } from '../../../../test/e2eTestUtils';

describe('gux-phone-input', () => {
  describe('#render', () => {
    [
      {
        description: 'should render phone-input',
        html: `
        <gux-phone-input-beta></gux-phone-input-beta>
        `
      },
      {
        description: 'should render with default region',
        html: `<gux-phone-input-beta default-region="US"></gux-phone-input-beta>`
      },
      {
        description: 'should render with value',
        html: `<gux-phone-input-beta value="+13175971660"></gux-phone-input-beta>`
      },
      {
        description: 'should render with format',
        html: `<gux-phone-input-beta phone-number-format="E164"></gux-phone-input-beta>`
      }
    ].forEach(({ description, html }) => {
      it(description, async () => {
        const page = await newSparkE2EPage({ html });
        const element = await page.find('gux-phone-input-beta');
        expect(element.outerHTML).toMatchSnapshot();
        await a11yCheck(page);
      });
    });

    it('open country dropdown and select option', async () => {
      const page = await newSparkE2EPage({
        html: `<gux-phone-input-beta></gux-phone-input-beta>`
      });
      const dropdownButton = await page.find('pierce/.gux-field-button');
      const inputField = await page.find('pierce/#tel-input');

      const inputEventSpy = await page.spyOnEvent('input');

      await dropdownButton.press('Enter');
      expect(dropdownButton.getAttribute('aria-expanded')).toBe('true');
      await page.waitForChanges();
      await a11yCheck(page);

      await page.keyboard.press('ArrowDown');
      await page.keyboard.press('Enter');
      await page.waitForChanges();

      expect(await inputField.getProperty('value')).toBe('+93');
      expect(inputEventSpy).toHaveReceivedEventDetail('+93');
      expect(dropdownButton.getAttribute('aria-expanded')).toBe('false');
    });
  });

  it('region is set when initialized with value', async () => {
    const page = await newSparkE2EPage({
      html: `<gux-phone-input-beta value="+13175971660"></gux-phone-input-beta>`
    });
    const dropdownButton = await page.find('pierce/.gux-field-button');
    await dropdownButton.click();
    await page.waitForChanges();

    const selected = await page.find('pierce/.gux-selected');

    expect(selected.textContent).toEqual('United States+1');
  });

  it('region is set when typing in country code', async () => {
    const page = await newSparkE2EPage({
      html: `<gux-phone-input-beta></gux-phone-input-beta>`
    });
    const dropdownButton = await page.find('pierce/.gux-field-button');
    const inputField = await page.find('pierce/#tel-input');

    await inputField.type('+41');
    await page.waitForChanges();

    await dropdownButton.click();
    await page.waitForChanges();

    const selected = await page.find('pierce/.gux-selected');

    expect(selected.textContent).toEqual('Switzerland+41');
  });

  it('phone validation error is sent as true', async () => {
    const page = await newSparkE2EPage({
      html: `<gux-phone-input-beta></gux-phone-input-beta>
            <input type="text" id="focus-input">`
    });
    const element = await page.find('gux-phone-input-beta');
    const inputField = await page.find('pierce/#tel-input');

    const validationEventSpy = await page.spyOnEvent('phoneValidationError');

    await inputField.focus();
    await inputField.type('+1317');
    await page.waitForChanges();
    element.triggerEvent('focusout');
    await page.waitForChanges();

    expect(validationEventSpy).toHaveReceivedEventDetail(true);
  });
});
