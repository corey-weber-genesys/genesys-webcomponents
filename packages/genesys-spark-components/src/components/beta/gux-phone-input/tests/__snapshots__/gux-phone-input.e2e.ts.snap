// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-phone-input #render should render phone-input 1`] = `"<gux-phone-input-beta hydrated=\\"\\"></gux-phone-input-beta>"`;

exports[`gux-phone-input #render should render with default region 1`] = `"<gux-phone-input-beta default-region=\\"US\\" hydrated=\\"\\"></gux-phone-input-beta>"`;

exports[`gux-phone-input #render should render with format 1`] = `"<gux-phone-input-beta phone-number-format=\\"E164\\" hydrated=\\"\\"></gux-phone-input-beta>"`;

exports[`gux-phone-input #render should render with value 1`] = `"<gux-phone-input-beta value=\\"+13175971660\\" hydrated=\\"\\"></gux-phone-input-beta>"`;
